x-pack/plugins/security_solution/public/management/cypress/e2e/artifacts.cy.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import {
+  ENDPOINT_ARTIFACT_LIST_IDS,
+  EXCEPTION_LIST_URL,
+} from '@kbn/securitysolution-list-constants';
+import { BASE_ENDPOINT_ROUTE } from '../../../../common/endpoint/constants';
+import { login, loginWithRole, ROLE } from '../tasks/login';
+
+import { type FormAction, getArtifactsListTestsData } from './mocks/artifacts_page';
+import { runEndpointLoaderScript } from '../tasks/run_endpoint_loader';
+
+const removeAllArtifacts = () => {
+  for (const listId of ENDPOINT_ARTIFACT_LIST_IDS) {
+    cy.request({
+      method: 'DELETE',
+      url: `${EXCEPTION_LIST_URL}?list_id=${listId}&namespace_type=agnostic`,
+      headers: { 'kbn-xsrf': 'kibana' },
+      failOnStatusCode: false,
+    });
+  }
+};
+
+const loginWithWriteAccess = (url: string) => {
+  loginWithRole(ROLE.analyst_hunter);
+  cy.visit(url);
+};
+
+const loginWithReadAccess = (url: string) => {
+  loginWithRole(ROLE.t2_analyst);
+  cy.visit(url);
+};
+
+const loginWithoutAccess = (url: string) => {
+  loginWithRole(ROLE.t1_analyst);
+  cy.visit(url);
+};
+
+// Checks for Endpoint data and creates it if needed
+const loadEndpointDataForEventFiltersIfNeeded = () => {
```

Comment:
same here. no async needed?

Reply:
Same comment as above, let me know if further explanation is needed 🙂

Reply:
ok. cool. thanks. I'll read up on it some more